app/src/ui/lib/theme-change-monitor.ts

```diff
@@ -0,0 +1,60 @@
+import { remote } from 'electron'
+import { ApplicationTheme } from './application-theme'
+import { IDisposable, Disposable, Emitter } from 'event-kit'
+import { supportsDarkMode, isDarkModeEnabled } from './dark-theme'
+
+class ThemeChangeMonitor implements IDisposable {
+  private readonly emitter = new Emitter()
+
+  private subscriptionID: number | null = null
+
+  public constructor() {
+    this.subscribe()
+  }
+
+  public dispose() {
+    this.unsubscribe()
+  }
+
+  private subscribe = () => {
+    if (!supportsDarkMode()) {
+      return
+    }
+
+    this.subscriptionID = remote.systemPreferences.subscribeNotification(
+      'AppleInterfaceThemeChangedNotification',
+      this.onThemeNotificationFromOS
+    )
+  }
+
+  private onThemeNotificationFromOS = (event: string, userInfo: any) => {
+    const darkModeEnabled = isDarkModeEnabled()
+
+    const theme = darkModeEnabled
+      ? ApplicationTheme.Dark
+      : ApplicationTheme.Light
+    this.emitThemeChanged(theme)
+  }
+
+  private unsubscribe = () => {
+    if (this.subscriptionID !== null) {
+      remote.systemPreferences.unsubscribeNotification(this.subscriptionID)
+    }
+  }
+
+  public onThemeChanged(fn: (theme: ApplicationTheme) => void): Disposable {
+    return this.emitter.on('theme-changed', fn)
+  }
+
+  private emitThemeChanged(theme: ApplicationTheme) {
+    this.emitter.emit('theme-changed', theme)
+  }
+}
+
+// this becomes our singleton that we can subscribe to from anywhere
+export const themeChangeMonitor = new ThemeChangeMonitor()
```

Comment:
In how many places do we have to subscribe to this object? If it's only in the `AppStore`, I think we can axe this and setup our single instance there.

Reply:
It is only currently used in the `AppStore`. Were you thinking we do something like `const themeChangeMonitor = new ThemeChangeMonitor()` in the constructor of `AppStore`? 

I can't currently think of a reason why we would want to watch monitor theme changes elsewhere but we at least get the ability to do so this way consistently to my knowledge 🤔